### ISSUE TITLE
MVK flagged as fail when it should be PASS

### ISSUE THREAD

**Issue by rljacob** [E3SM-Project/E3SM#5695]
THe last time MVK passed (April 13 https://my.cdash.org/test/77934058?graph=status), this was the output at the bottom:
```
 ---------------------------------------------------
2023-04-13 11:32:44: BASELINE PASS for test 'JNextAtm_nbfb20230413_003526'.
    Test status: pass; Variables analyzed: 121; Rejecting: 9; Critical value: 13; Ensembles: statistically identical EVV results can be viewed at:
    https://web.lcrc.anl.gov/public/e3sm/e3smtest/evv/MVK_PS.ne4_oQU240.F2010.chrysalis_intel.C.JNextAtm_nbfb20230413_003526/index.html
 ---------------------------------------------------
2023-04-13 11:32:44: TPUTCOMP: Computation time changed by 12.88% relative to baseline
 ---------------------------------------------------
2023-04-13 11:32:44: Error: TPUTCOMP: Computation time increase > 5% from baseline
 ---------------------------------------------------
```

The next day, April 14, it failed even though the actual test still said "BASELINE PASS".  This was the output.

```
 ---------------------------------------------------
2023-04-14 02:11:53: BASELINE PASS for test 'JNextAtm_nbfb20230414_003443'.
    Test status: pass; Variables analyzed: 121; Rejecting: 8; Critical value: 13; Ensembles: statistically identical
    EVV results can be viewed at:
        https://web.lcrc.anl.gov/public/e3sm/e3smtest/evv/MVK_PS.ne4_oQU240.F2010.chrysalis_intel.C.JNextAtm_nbfb20230414_003443/index.html
 ---------------------------------------------------
2023-04-14 02:11:53: TPUTCOMP: Computation time changed by 4.18% relative to baseline
 ---------------------------------------------------
2023-04-14 02:11:53: memleak detected, memory went from 2003.380000 to 2129.530000 in 10126 days
 ---------------------------------------------------
2023-04-14 02:11:53: NEW FAIL: Potentially broken merges:
a1732bdf7bb39eee8f2b0693b63f2965d1bf9067
a6646afed0010843691d0424c7968c9a842c3cfa
8d4e8e5a91ddf67946e87336cc8ed9b76af36779
a3becd7f5ed38e5e454b443f10a800114026d959
 ---------------------------------------------------
```

Apparently the "memleak detected" is causing a FAIL but I thought we didn't allow that.  @amametjanov ?

**Comment by jgfouca** [comment]
@rljacob ,  it depends on the flags given to our jenkins script. It defaults to not checking memory. Looking through our jenkins scripts, I see this:

```
% git grep check
jenkins/chrysalis_prod_bench.sh:$RUNSCRIPT -t e3sm_prod_bench -O master --baseline-compare --check-throughput --check-memory --pes-file $E3SMREPO/cime_config/testmods_dirs/config_pes_tests.xml
```

So, the chysalis_prod script is turning on memory checking.

**Comment by rljacob** [comment]
what about the e3sm_atm_nbfb script?

**Comment by jgfouca** [comment]
@rljacob , sorry for the confusion. I thought maybe the case above was being run by the chrysalis_prod script but it looks like that is not the case. memleak should not be causing fails in any other test jobs.

**Comment by amametjanov** [comment]
`memleak detected` is a fail within the test: this check is always on.
`Memory usage increase >N% from baseline` is a baseline comparison fail: this check is off by default (on with `--check-memory`).

**Comment by jgfouca** [comment]
Thanks @amametjanov ! You're right, I got mixed up. We have `check_memory` for checking against baseline memory usage and `ignore_memleak` for ignoring memleaks which is off by default (so memleaks are not ignored by default).

**Comment by mkstratos** [comment]
All the fails I can see that aren't statistical have `memleak detected`, and the memleak percentage has been between 5.5 - 8.5% or so since February (when present). The bigger percentages seem to be recent, but don't seem related to the climate changing PR. Do we need to address a real memory leak, or would `ignore_memleak` be sufficient, since it's occasional and relatively small?

![plt_memleak_hist](https://github.com/E3SM-Project/E3SM/assets/1476361/d049cfff-a2f1-4c85-9f7e-fdb804caa851)
From CDash data, memleak percentage time series and histogram.

**Comment by rljacob** [comment]
Remind me how the MVK test does a model run?  Its using multi-instance right?

**Comment by mkstratos** [comment]
Yes, it's a 30 instance ensemble

**Comment by sarats** [comment]
My 2 cents, Let's start with `ignore_memleak` to make the overall test usable while we figure out if we wish to debug this further.

**Comment by rljacob** [comment]
Yes lets get it passing again.  We should also create a new baseline since after #5687 was merged its still showing a fail (for the actual test) by 2 variables over the limit.

**Comment by rljacob** [comment]
You need to add something to CIME?  There is already a --ignore-memleak option in create_test.  Don't we just add that to the script to run this suite?

**Comment by jgfouca** [comment]
@rljacob The option needs to be passed down from jenkins_generic_job.

**Comment by rljacob** [comment]
@amametjanov remind me what that memleak test is monitoring to determine a leak.  Is the mem diags from the coupler?

**Comment by jgfouca** [comment]
@rljacob , it looks like it searches for this regex in the cpl log:

```
.*model date =\s+(\w+).*memory =\s+(\d+\.?\d+).*highwater"
```

**Comment by jgfouca** [comment]
If the value above increases by more than `TEST_MEMLEAK_TOLERANCE` (defaults to 10%) from the original date to the final date of the simulation, then the test is marked as MEMLEAK.

**Comment by rljacob** [comment]
That's an xml variable so we could add testmod to increase it but lets go with turning it off for now.

**Comment by rljacob** [comment]
@mkstratos from your figure in https://github.com/E3SM-Project/E3SM/issues/5695#issuecomment-1555106084, there's definitely more variance in the memory after April 16.  But I'm wondering why there's any variance?  If the test run's the same way each time, wouldn't the memory increase be the same?

**Comment by rljacob** [comment]
Unless code changes were making it change like that.

**Comment by mkstratos** [comment]
My assumption is that because the ensemble is unique for each day / test, there may be some variance in memory.

**Comment by jgfouca** [comment]
I have changed jenkins to accept --ignore-memleak and changed the script specific to this job to use this flag. Should I close the issue?

**Comment by rljacob** [comment]
Yes you can.   Today's MVK test on chrysalis is pass even though this message is in the TestStatus.log:

> 2023-05-23 08:52:13: memleak detected, memory went from 2033.890000 to 2164.010000 in 10126 days

That's a 6% increase but TEST_MEMLEAK_TOLERANCE on chrysalis is set to .05 so would have failed without that mod.
